pandas/core/generic.py

```diff
@@ -2149,13 +2149,13 @@ def to_excel(self, excel_writer, sheet_name="Sheet1", na_rep="",
 
         df = self if isinstance(self, ABCDataFrame) else self.to_frame()
 
-        max_rows = 2**20
-        max_cols = 2**14
+        excel_max_rows = 2**20
+        excel_max_cols = 2**14
         num_rows, num_cols = df.shape
-        if num_rows > max_rows or num_cols > max_cols:
+        if num_rows > excel_max_rows or num_cols > excel_max_cols:
             raise ValueError(f"This sheet is too large! Your sheet size is: "
                              f"{(num_rows, num_cols)}. "
-                             f"Max sheet size is: {(max_rows, max_cols)}.")
+                             f"Max sheet size is: {(excel_max_rows, excel_max_cols)}.")
```

Comment:
so all of this code checking the size should be moved to the ExcelFormatter

Reply:
yep! I made the change locally, but I'm waiting to push. I'm currently ensuring my new tests work, but I'm running into an issue. 

When running:
     pytest pandas/tests/io/test_excel.py

I receive the following error:
    ('DeprecationWarning', DeprecationWarning('The html argument of XMLParser() is deprecated'

I removed all my changes and still run into the issue.

Reply:
Try upgrading xlrd. That deprecation warning is new to 3.7 so maybe if you have an older version of xlrd doesn't handle that yet

Reply:
my environment is running xlrd: 1.2.0. I ran conda update and no updates were available. Still running into the same issue.

Reply:
Hmmm. I rolled back to the latest commit that was accepted to master.

commit 22c2b73857d1035685d42b0df56f4762ffad0611 (HEAD -> rollback_pandas, origin/master, origin/HEAD, master)
Author: Gregory Rome <gwrome@gmail.com>
Date:   Fri Apr 12 09:46:18 2019 -0500

    Fix type annotations in pandas.core.dtypes (#26029)

I'm running the same command and running into the same issue.

Reply:
Hmm ok well just push what you have will be easier to take a look that way

Reply:
pushed.